worlds/hades/Client.py

```diff
@@ -0,0 +1,409 @@
+                if len(collect_locations_cache) > 0:
+                    collect_locations_cache = collect_locations_cache[:-1]
+                    subsume.Send(styx_scribe_send_prefix + "Locations collected:" + collect_locations_cache)
+
+        if cmd == "ReceivedItems":
+            # What should be done when an Item is recieved.
+            # NOTE THIS GETS ALL ITEMS THAT HAVE BEEN RECIEVED! WE USE THIS FOR RESYNCS!
+            for item in args["items"]:
+                self.cache_items_received_names += [self.item_names.lookup_in_slot(item.item)]
+
+            # We ignore sending the package to hades if just connected,
+            # since the game is not ready for it (and will request it itself later)
+            if self.is_receiving_items_from_connect_package:
+                return
+            self.send_items()
+
+        if cmd == "LocationInfo":
+            if self.creating_location_to_item_mapping:
+                self.creating_location_to_item_mapping = False
+                asyncio.create_task(self.create_location_to_item_dictionary(args["locations"]))
+                return
+            super().on_package(cmd, args)
+            
+        if cmd == "Bounced":
+            if "tags" in args:
+                if "DeathLink" in args["tags"]:
+                    self.on_deathlink(args["data"])
+
+    def send_items(self):
+        payload_message = self.parse_array_to_string(self.cache_items_received_names)
+        subsume.Send(styx_scribe_send_prefix + "Items Updated:" + payload_message)
+
+    def parse_array_to_string(self, array_of_items):
+        message = ""
+        for itemname in array_of_items:
+            message += itemname
+            message += ","
+        return message
+
+    async def send_location_check_to_server(self, message):
+        sendingLocationsId = []
+        sendingLocationsName = message
+        payload_message = []
+        sendingLocationsId += [self.location_name_to_id[sendingLocationsName]]
+        payload_message += [{"cmd": "LocationChecks", "locations": sendingLocationsId}]
+        asyncio.create_task(self.send_msgs(payload_message))
+
+    async def check_connection_and_send_items_and_request_starting_info(self, message):
+        if self.check_for_connection():
+            self.is_receiving_items_from_connect_package = False
+            await self.send_items_and_request_starting_info(message)
+
+    async def send_items_and_request_starting_info(self, message):
+        self.store_settings_data()
+        # send items that were already cached in connect
+        self.send_items()
+        self.request_location_to_item_dictionary()
+
+    def store_settings_data(self):
+        hades_settings_string = ""
+        #codify in the string all heat settings
+        hades_settings_string += str(self.hades_slot_data["heat_system"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["hard_labor_pact_amount"]) + "-"
```

Comment:
You could shrink this a little with something like
```py
"-".join(str(setting for setting in [
    "heat_system",
    "hard_labor_pact_amount",
])
```
The polycosmos version throws a small wrench in it. And if you rely on a trailing `-`, watch out for that.